jsonrpc/eth_endpoint.go

```diff
@@ -27,11 +27,16 @@ type ethTxPoolStore interface {
 	GetPendingTx(txHash types.Hash) (*types.Transaction, bool)
 }
 
+type Account struct {
+	Balance *big.Int
+	Nonce   uint64
+}
+
 type ethStateStore interface {
-	GetAccount(root types.Hash, addr types.Address) (*state.Account, error)
+	GetAccount(root types.Hash, addr types.Address) (*Account, error)
 	GetStorage(root types.Hash, addr types.Address, slot types.Hash) ([]byte, error)
 	GetForksInTime(blockNumber uint64) chain.ForksInTime
-	GetCode(hash types.Hash) ([]byte, error)
+	GetCode(root types.Hash, addr types.Address) ([]byte, error)
```

Comment:
Is it get contract code?

Reply:
The `GetCode` endpoint (eth_getCode) returns the code for an address. Thus, first, it resolves the address, retrieves the code hash, and then resolves the code (since the code is internally stored as `codeHash:code`)

Reply:
I just wanted to mention that to return contract code you need code hast(it's unique) or account(to get code hash). I dont understand well why do you need both of them.

Reply:
Following [this](https://ethereum.org/en/developers/docs/apis/json-rpc/#eth_getcode) spec, you can get the code at some point in time. That root is used to resolve the `Account` at some point in time.

Reply:
Does it makes sense to reverse the param order as in the spec?